laa_court_data_api_app/internal/court_data_adaptor_client.py

```diff
@@ -22,13 +22,23 @@ async def get(self, endpoint: str, params: Optional[dict[str, str]] = None,
     async def post(self, endpoint: str, params: Optional[dict[str, str]] = None,
                    headers: Optional[dict[str, any]] = None,
                    body: Optional[any] = None):
+        if headers is None:
+            headers = {}
+        headers.update({"Content-Type": "application/json"})
+        if body is not None:
+            body = body.json()
```

Comment:
What we are doing here is converting the model from its type to a json structure. Reason being httpx utilises the json library and this doesnt support non native types and will throw. By using pydantic to do this for us when the body exists we know it will be converted as we expect. The header above we pass across is to let the service we are sending this body to know that its a json structure